src/cloudsc_mix/CMakeLists.txt

```diff
@@ -0,0 +1,139 @@
+# (C) Copyright 1988- ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+# only double-precision currently supported
+if( "${prec}" STREQUAL "dp" )
+
+    # Define integer IDs corresponding to every language choice
+    ## kernel language
+    set(CUDA_LANG "1")
+    set(HIP_LANG "2")
+    set(SYCL_LANG "3")
+    ## offload model
+    set(ACC_OFFLOAD "1")
+    set(OMP_OFFLOAD "2")
+
+    # Select offload model
+    if( HAVE_ACC )
+        set( INIT_GPU_OFFLOAD "ACC" )
+    elseif( HAVE_OMP )
+        set( INIT_GPU_OFFLOAD "OMP" )
+    else()
+        set( INIT_GPU_OFFLOAD "UNDEFINED" )
+    endif()
+
+    if ( NOT DEFINED(CLOUDSC_MIX_GPU_OFFLOAD ))
+      set( CLOUDSC_MIX_GPU_OFFLOAD ${INIT_GPU_OFFLOAD} CACHE STRING "The programming model to use for data offload in mix variants (ACC or OMP)" )
+    endif()
```

Comment:
NB: The conditional around this is redundant: If the variable is defined in the cache, the existing cache value is used.